src/Xrefcheck/Scanners/Markdown.hs

```diff
@@ -167,12 +174,46 @@ nodeExtractInfo input@(Node _ _ nSubs) = do
                   [t]    -> (t, Nothing)
                   t : ts -> (t, Just $ T.intercalate "#" ts)
                   []     -> error "impossible"
-          return $ FileInfoDiff
-            (DList.singleton $ Reference {rName, rPos, rLink, rAnchor})
-            DList.empty
+          return mempty
+            { _fidReferences = DList.singleton $ Reference {rName, rPos, rLink, rAnchor}
+            }
 
         _ -> return mempty
 
+    copyPaste :: Node -> m FileInfoDiff
+    copyPaste (Node _ (LIST _) nodes) = do
+      case items of
+        top : rest | urlIsASubsequence top -> do
+          let bad = filter (not . urlIsASubsequence) rest
+          pure mempty { _fidCopyPastes = DList.fromList bad }
+        _ -> do
+          pure mempty
+      where
+        items = do
+          (_, nodes',  _) <- takeOnly _ITEM nodes
+          (_, nodes'', _) <- takeOnly _PARAGRAPH nodes'
+          take 1 $ do
+            (_, texts, (url, _)) <- takeOnly _LINK nodes''
+            (pos, _, txt) <- take 1 $ takeOnly _TEXT texts
+            return (CopyPaste url txt (toPosition pos))
+
+    copyPaste _ = pure mempty
+
+    takeOnly prizm list = do
+      Node pos hdr nodes <- list
+      case hdr^?prizm of
+        Just res -> return (pos, nodes, res)
+        Nothing  -> []
+
+urlIsASubsequence :: CopyPaste -> Bool
```

Comment:
**Nitpick:** AFAICS, here we check if link text is a subsequence of link URL, so __if name is a URL subsequence__, not __URL is a subsequence of name__